# Blockage ratio/ Equivalence ratio/ Angle for LBO limit

## verification for gri-mech 211

### 调试（cfd燃烧温度偏高）
- (refer to paper) external emissivtity = 0.7 -> 0.8 (the larger value means that more easy for heat transfering)
- 将interl emissivity = 0.8 -》1 (并不关注这个)
- (refer to paper) 修改固体(steeless)导热率（refer to paper）
- (refer to paper -{Jiang, 2013 #849}) 空气对流系数10 w/(m2·K), emissivity = 0.7-0.8
- 将空气对流系数改为20 w/(m2·K)，emissivity = 0.85;

In [12]:
t_exp = 1250
t_cfd = 1450
relative_error = (t_cfd-t_exp)/t_exp 
relative_error
#最大误差14.72%

0.16

### 重要算例文件保存

-  二维轴对称，不开辐射模型，计算稳态，计算结果良好："E:/John_Mike/blockage_ratio/blockage_files/dp0/FLU-1/Fluent/SYS-1-Setup-steady-ok.dat.gz"
- 待算：将70ms那个算例patch一个流体域的全局高温，计算瞬态

## Blockage ratio compute

In [13]:
180/1250

100/1350

200/1250

0.16

## 工况计算

In [ ]:
print('''
针对长方体式的非预混微型燃烧器
              =======================================
   air-inlet->|___->                                |
               ___|                                 |
  fuel-inlet->|___->    2-D Micro Combustor         |
               ___|                                 |
   air-inlet->|   ->                                |
              =======================================
''')

# unit m2 流通截面积 (根据几何模型参数计算【单位：mm】, 注意结果单位需由mm2转换为m2)

cross_area_air = 2 * 1 * 1.75 / 10**(-6) # 2 为左右两个空气入口， 1 为高度，1.75为宽度
cross_area_fuel = 1 * 0.3 / 10**(-6) # 0.3 为中心燃料入口宽度

# unit kg/m3 密度 (通用)
density_ch4 = 0.65
density_air = 1.23
density_h2 = 0.09

# 对于CH4-H2混合燃料而言，求特定当量比和掺氢比下的空气流速，包括定当量比，变掺氢比与定掺氢比，变当量比两种选择

print("-----定当量比,变掺氢比,求空气流速-----")
print("-----定当量比,变掺氢比,求空气流速-----")
computation_numbers = int(input("[in]请输入接下来要计算多少组数据(多少个不同工况下对应的空气流速):"))
velocity_fuel = float(input("[in]请输入燃料入口流速(m/s):"))

# 选择d：定当量比，变掺氢比，求对应的空气流速

equivalence_ratio = float(input("[in]请输入定当量比的值:"))
for i in range(1, computation_numbers + 1):
    xh_2 = float(input("[in]请输入此次变掺氢比的值(以小数表示):"))

    # air to fuel stoic of ch4 & h2 mixture fuel
    # 下面一行的详细说明见OneNote笔记"掺氢燃烧二维模型尺寸-文章"
    # 主要解决的是由掺氢带来的理论当量比的变化; 对于纯甲烷燃料燃烧; 这是一个定值, 而对于甲烷掺氢混合燃料而言, 该值将随着掺氢比的变化而变化, 即与掺氢比有关, 关系式如下:
    air_to_fuel_stoic = 137.25 * (2 - 1.5 * xh_2) / (16 - 14 * xh_2)

    # velocity of air supply
    velocity_air = (air_to_fuel_stoic / equivalence_ratio) * (
                (1 - xh_2) * density_ch4 * velocity_fuel + xh_2 * density_h2 * velocity_fuel) * cross_area_fuel / (
                               density_air * cross_area_air)
    print(
        f"[output]第{i}组,已知:定当量比={equivalence_ratio}, 变掺氢比={xh_2}; 此时计算的空气流速大小为:{velocity_air:4f}m/s")

print(f"[over]定当量比={equivalence_ratio}，空气流速计算结束")

# 保持窗口处于待命状态，退出/结束程序需键入回车指令：键入Enter

input("Press Enter to exit ;)")


针对长方体式的非预混微型燃烧器
   air-inlet->|___->                                |
               ___|                                 |
  fuel-inlet->|___->    2-D Micro Combustor         |
               ___|                                 |
   air-inlet->|   ->                                |

-----定当量比,变掺氢比,求空气流速-----
-----定当量比,变掺氢比,求空气流速-----
[in]请输入接下来要计算多少组数据(多少个不同工况下对应的空气流速):1
[in]请输入燃料入口流速(m/s):20
[in]请输入定当量比的值:0.6
[in]请输入此次变掺氢比的值(以小数表示):0
[output]第1组,已知:定当量比=0.6, 变掺氢比=0.0; 此时计算的空气流速大小为:25.903746m/s
[over]定当量比=0.6，空气流速计算结束


计算工况

| Fuel inlet velocity(m/s) | equivalence ratio(LBO-1.4) | air inlet velocity(m/s) |  Blockage ratio(0.4-0.8,0.2)    |      |
| ------------------------ | -------------------------- | --------------------------- | ---- | ---- |
|         10                 |        0.6                    |        12.95       |   |      |
|         10                 |        1.0                    |        7.77       |   |      |
|         20              |           0.6                 |           25.9                  |      |      |
|                          |                            |                             |      |      |
|                          |                            |                             |      |      |
|                          |                            |                             |      |      |
|                          |                            |                             |      |      |

## 计算blockage ratio的合适取值及范围以及与L_4的关系

In [37]:
0.35-0.125

0.22499999999999998

In [19]:
(0.85-0.35)/4

0.125

In [20]:
0.35+0.125

0.475

In [21]:
0.35+0.125*2

0.6

In [22]:
0.35+0.125*3

0.725

In [23]:
0.35+0.125*4

0.85

In [38]:
# 已知预设值blockage ratio(BR) = xx，求此时的长度L_4(mm)

L_2 = 0.2
W_1 = 4.4
blockage_ratio = 0.225

L_4 = blockage_ratio * (W_1 - 2 * L_2) /2

print(L_4)

0.45


Blockage ratio

| BR(预设值) | 对应调整L4长度为(mm)(建模时需减去0.1mm) |
| ---------- | --------------------------------------- |
| 0.225      | 0.45                                    |
| 0.35       | 0.7                                     |
| 0.475      | 0.95                                    |
| 0.6        | 1.2                                     |
| 0.725x     | 1.45x                                   |
| 0.85x      | 1.7x                                    |

### 计算AR

In [36]:
360-280

80

175/1250

In [ ]:
175/1250